server/app/category.go

```diff
@@ -130,3 +134,109 @@ func (a *App) DeleteCategory(categoryID, userID, teamID string) (*model.Category
 
 	return deletedCategory, nil
 }
+
+func (a *App) moveBoardsToDefaultCategory(userID, teamID, sourceCategoryID string) error {
+	// we need a list of boards associated to this category
+	// so we can move them to user's default Boards category
+	categoryBoards, err := a.GetUserCategoryBoards(userID, teamID)
+	if err != nil {
+		return err
+	}
+
+	var sourceCategoryBoards *model.CategoryBoards
+	defaultCategoryID := ""
+
+	// iterate user's categories to find the source category
+	// and the default category.
+	// We need source category to get the list of its board
+	// and the default category to know its ID to
+	// move source category's boards to.
+	for i := range categoryBoards {
```

Comment:
The way you are looping here is confusing to me, I don't know exactly what you need to use the index instead of a copy to the original data. For me is way more readable something like:

```go
	for _, category := range categoryBoards {
		if category.ID == sourceCategoryID {
			sourceCategoryBoards = &category
		}

		if category.Name == defaultCategoryBoards {
			defaultCategoryID = category.ID
		}

		// if both categories are found, no need to iterate furthur.
		if sourceCategoryBoards != nil && defaultCategoryID != "" {
			break
		}
	}
```

Reply:
the other version makes me think you need to work with references to modify the same memory for some reason, so It was confusing to me.

Reply:
I do nee to work with references.

```go
if category.ID == sourceCategoryID {
			sourceCategoryBoards = &category
		}
```

Here I need to save the reference of categoryboard while iterating.